pkg/controller/directimagemigration/imagestream.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2020 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package directimagemigration
+
+import (
+	"context"
+	//  liberr "github.com/konveyor/controller/pkg/error"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	imagev1 "github.com/openshift/api/image/v1"
+	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) getDirectImageStreamMigrations() ([]migapi.DirectImageStreamMigration, error) {
+	ismList := migapi.DirectImageStreamMigrationList{}
+	err := t.Client.List(
+		context.TODO(),
+		k8sclient.MatchingLabels(t.Owner.GetCorrelationLabels()),
+		&ismList)
+	if err != nil {
+		return nil, err
+	}
+	return ismList.Items, nil
+}
+
+func (t *Task) createDirectImageStreamMigrations() error {
+	// Get client for source
+	srcClient, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+	// Get list namespaces to iterate over
+	for srcNsName, destNsName := range t.Owner.GetNamespaceMapping() {
+		isList := imagev1.ImageStreamList{}
```

Comment:
I wonder if we should separate this list call from this method. Since this is `a one shot` run controller, we would want to have separate phases for:
1. Getting the list of imagestreams we commit to migrating
2. Reading the committed list and create dism resources for them

This is to prevent the scenario where we error out creating one or more dism and during the next reconcile loop we find more resources then the initial reconcile loop

Reply:
+1 on the rework, reads very nice and seems more declarative